### PR TITLE
Ratings comments UI for Store

### PR DESCRIPTION
**Pull request by msm1992**
### Proposed changes in this pull request
Star Ratings Bar 
screenshot from 2017-09-25 11-23-11

Comments
screenshot from 2017-09-25 11-23-28

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ class BasicInfo extends Component {
         };
         this.api_uuid = this.props.uuid;
         this.logChange = this.logChange.bind(this);
+       // this.handleStarRatingUpdate = this.handleStarRatingUpdate(this);
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js]: Shall we remove the commented code line?

Comment (msm1992) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js]: removed

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/data/api.js [hunk 2]
```diff
@@ -248,6 +248,85 @@ class API {
...
+    */
+    addComment(api_id, commentInfo, callback = null) {
+        var promise = this.client.then(
+            (client) => {
+                return client.apis["API (individual)"].post_apis__apiId__comments(
+                    {apiId: api_id, body: commentInfo}, this._requestMetaData());
+            }
+        ).catch(
+             error => {
+                 alert(error);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/data/api.js]: are we using alerts. Any alternatives? @chanaka3d

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/data/api.js [hunk 3]
```diff
@@ -248,6 +248,85 @@ class API {
...
+    addRating(api_id, ratingInfo, callback = null) {
+        var promise = this.client.then(
+            (client) => {
+                console.log(client.apis["API (individual)"]);
+                return client.apis["API (individual)"].put_apis__apiId__user_rating(
+                    {apiId: api_id, body: ratingInfo}, this._requestMetaData());
+            }
+        ).catch(
+             error => {
+                 alert("error in adding "+error);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/data/api.js]: need spaces around  +

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/Overview.js [hunk 4]
```diff
@@ -148,6 +155,45 @@ class Overview extends Component {
...
+
+    handleGetAllComments() {
+        var api = new Api();
+        let promise_get = api.getAllComments(this.api_uuid);
+        promise_get.then(
+            response => {
+                var index = 0;
+                var comments = [];
+                this.setState({commentList : response.obj.list});
+                console.log(comments);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/Overview.js]: can we remove consoles logs and use proper logs available in react?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js [hunk 5]
```diff
@@ -375,4 +380,97 @@ class BasicInfo extends Component {
...
+            response => {
+                console.info(response.obj);
+            }
+        );
+
+        //get user rating
+        let promised_rating = api.getRatingFromUser(this.props.apiIdProp, null);
+        promised_rating.then(
+            response => {
+                console.log(response.obj.userRating);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js]: avoid consoles logs

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js [hunk 6]
```diff
@@ -375,4 +380,97 @@ class BasicInfo extends Component {
...
+        this.handleRatingUpdate = this.handleRatingUpdate.bind(this);
+        this.handleMouseOut = this.handleMouseOut.bind(this);
+    }
+
+    componentDidMount() {
+        var api = new Api();
+        let promised_api = api.getAPIById(this.props.apiIdProp);
+        promised_api.then(
+            response => {
+                console.info(response.obj);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js]: avoid console logs

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js [hunk 7]
```diff
@@ -375,4 +380,97 @@ class BasicInfo extends Component {
...
+            <span onMouseOver = {this.handleHoveringOver} style={{color: 'gold'}}>
+                ☆
+            </span>;
+    }
+}
+
+class StarRatingBar extends React.Component {
+    constructor(props) {
+        super(props);
+
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js]: do we need space here?

**Pull request merged**